src/qibolab/designs/qmrs.py

```diff
@@ -0,0 +1,160 @@
+
+        self.is_connected = False
+
+        # Instantiate QM OPX instruments
+        if simulation_duration is None:
+            from qibolab.instruments.qm import QMOPX
+
+            self.opx = QMOPX("qmopx", address)
+        else:
+            from qibolab.instruments.qmsim import QMSim
+
+            self.opx = QMSim("qmopx", address, simulation_duration, cloud)
+
+        # Map controllers to qubit channels (HARDCODED)
+        # readout
+        Channel("L3-25_a").ports = [
+            ("con1", 10),
+            ("con1", 9),
+        ]
+        Channel("L3-25_b").ports = [
+            ("con2", 10),
+            ("con2", 9),
+        ]
+        # feedback
+        Channel("L2-5").ports = [("con1", 1), ("con1", 2)]
+        # drive
+        Channel("L3-11").ports = [("con1", 1), ("con1", 2)]
+        Channel("L3-12").ports = [("con1", 3), ("con1", 4)]
+        Channel("L3-13").ports = [("con1", 5), ("con1", 6)]
+        Channel("L3-14").ports = [("con1", 7), ("con1", 8)]
+        Channel("L3-15").ports = [("con3", 1), ("con3", 2)]
+        # flux
+        Channel("L4-1").ports = [("con2", 1)]
+        Channel("L4-2").ports = [("con2", 2)]
+        Channel("L4-3").ports = [("con2", 3)]
+        Channel("L4-4").ports = [("con2", 4)]
+        Channel("L4-5").ports = [("con2", 5)]
+
+        # Instantiate local oscillators (HARDCODED)
+        self.local_oscillators = [
+            SGS100A("lo_readout_a", "192.168.0.39"),
+            SGS100A("lo_readout_b", "192.168.0.31"),
+            SGS100A("lo_drive_low", "192.168.0.32"),
+            SGS100A("lo_drive_mid", "192.168.0.33"),
+            SGS100A("lo_drive_high", "192.168.0.34"),
+        ]
+
+        # Map LOs to channels
+        Channel("L3-25_a").local_oscillator = self.local_oscillators[0]
+        Channel("L3-25_b").local_oscillator = self.local_oscillators[1]
+        Channel("L3-15").local_oscillator = self.local_oscillators[2]
+        Channel("L3-11").local_oscillator = self.local_oscillators[2]
+        Channel("L3-12").local_oscillator = self.local_oscillators[3]
+        Channel("L3-13").local_oscillator = self.local_oscillators[4]
+        Channel("L3-14").local_oscillator = self.local_oscillators[4]
+
+        # Set default LO parameters in the channel
+        Channel("L3-25_a").lo_frequency = 7_300_000_000
+        Channel("L3-25_b").lo_frequency = 7_900_000_000
+        Channel("L3-15").lo_frequency = 4_700_000_000
+        Channel("L3-11").lo_frequency = 4_700_000_000
+        Channel("L3-12").lo_frequency = 5_600_000_000
+        Channel("L3-13").lo_frequency = 6_500_000_000
+        Channel("L3-14").lo_frequency = 6_500_000_000
+
+        Channel("L3-25_a").lo_power = 18.0
+        Channel("L3-25_b").lo_power = 15.0
+        Channel("L3-15").lo_power = 16.0
+        Channel("L3-11").lo_power = 16.0
+        Channel("L3-12").lo_power = 16.0
+        Channel("L3-13").lo_power = 16.0
+        Channel("L3-14").lo_power = 16.0
```

Comment:
These proposals are interesting, particularly because when scaling to more qubits we will be forced to use loops, instead of repeating everything. That being said, these are again hard-coded values which we could store somewhere and import here, like the IP address above.